test/spec/algoliasearch.helper/queryID.js

```diff
@@ -47,13 +48,13 @@ test('the response handler should check that the query is not outdated', functio
     helper: helper
   }];
 
-  helper._dispatchAlgoliaResponse(states, helper._lastQueryIdReceived + 1, null, testData.response);
-  helper._dispatchAlgoliaResponse(states, helper._lastQueryIdReceived + 10, null, testData.response);
+  helper._dispatchAlgoliaResponse(states, helper._lastQueryIdReceived + 1, testData.response);
```

Comment:
There is a test missing (even before this PR) with the behavior of the outdated query when the search returns errors. Since now the implementation is split into two parts, it's more obvious. We should test it.

Reply:
Gotcha!